x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/api.ts

```diff
@@ -33,37 +36,101 @@ import {
 } from './types';
 import { KibanaServices } from '../../../../common/lib/kibana';
 import * as i18n from '../../../pages/detection_engine/rules/translations';
+import { RulesSchema, rulesSchema } from '../../../../../common/detection_engine/schemas/response';
+import { validateEither } from '../../../../../common';
+import {
+  createRulesSchema,
+  updateRulesSchema,
+  patchRulesSchema,
+} from '../../../../../common/detection_engine/schemas/request';
+import { toError, toPromise } from '../../../../../common/shared_imports';
+
+/**
+ * Create provided Rule
+ *
+ * @param rule CreateRulesSchema to add
+ * @param signal to cancel request
+ *
+ * @throws An error if response is not OK
+ */
+const createRule = async ({ rule, signal }: CreateRulesProps): Promise<RulesSchema> =>
+  KibanaServices.get().http.fetch<RulesSchema>(DETECTION_ENGINE_RULES_URL, {
+    method: 'POST',
+    body: JSON.stringify(rule),
+    signal,
+  });
+
+const createRuleWithValidation = async ({ rule, signal }: CreateRulesProps): Promise<RulesSchema> =>
```

Comment:
Is that idea that we're validating on the FE and on the BE? Does that buy us anything to be able to validate from the UI?

Reply:
@peluja1012 Any opinions on this? IMO it's simpler, more readable, and more maintainable if we only validate in one place (the backend APIs).

Reply:
We still get the benefit of using common types on both API and UI code... that buys us a lot in terms of static type consistency. Once we start doing dynamic validation though, I feel like we're starting to hit the other side of the cost/benefit curve.

Reply:
I decided to remove these here for now. I think you bring up a lot of valid points that can be addressed separately if we decide to go down that route.